File: LeetCode Problems/1 to 100/1. Two Sum/1. Two Sum.java
class Solution {
    class Pair {
        int first, second;
        Pair(int a, int b) {
            first = a;
            second = b;
        }
    }
    public int[] twoSum(int[] nums, int target) {
        ArrayList<Pair> arr = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) arr.add(new Pair(nums[i], i));

        Collections.sort(arr, Comparator.comparing(p -> p.first));
        int l = 0, r = nums.length - 1;
        while (l <= r) {
            int sum = arr.get(l).first + arr.get(r).first;
            if (sum == target) {
                return new int[] {arr.get(l).second, arr.get(r).second};
            } else if (sum < target)
                l++;
            else
                r--;
        }
        return new int[] {-1, -1};
    }
}

File: LeetCode Problems/1 to 100/1. Two Sum/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        vector<pair<int, int>> arr;
        for (int i = 0; i < nums.size(); i++)
            arr.emplace_back(nums[i], i);
        sort(arr.begin(), arr.end());

        int l = 0, r = nums.size() - 1;
        while (l <= r) {
            int sum = arr[l].first + arr[r].first;
            if (sum == target) {
                return {arr[l].second, arr[r].second};
            } else if (sum < target)
                l++;
            else
                r--;
        }
        return {-1, -1};
    }
};
